packages/orchestrator-ui-components/src/rtk/api.ts

```diff
@@ -0,0 +1,35 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { graphqlRequestBaseQuery } from '@rtk-query/graphql-request-base-query';
+
+export enum BaseQueryTypes {
+    fetch = 'fetch',
+    graphql = 'graphql',
+    custom = 'custom',
+}
+
+type ExtraOptions = {
+    baseQueryType?: BaseQueryTypes;
+};
+
+export const orchestratorApi = createApi({
+    reducerPath: 'orchestratorApi',
+    baseQuery: (args, api, extraOptions: ExtraOptions) => {
+        const { baseQueryType } = extraOptions || {};
+
+        switch (baseQueryType) {
+            case BaseQueryTypes.fetch:
+                const fetchFn = fetchBaseQuery({
+                    baseUrl: 'https://orchestrator.dev.automation.surf.net/api',
```

Comment:
Try to use the config values, otherwise system-testing might be hard after merging this.
Same for the `default` case below